<center>Mateus Braga de Melo</center>


In [2]:
import pandas as pd

1ª Tarefa<br>
Carregue em dataframes os conjuntos de dados referentes ao ano de 2018, 2019 e 2020. Corrija quaisquer erros que apareçam<br> no carregamento dos arquivos. É possível acessar os arquivos<br>
compactados diretamente do site, sem precisar baixá-los?


Carregando DataFrames
<h3>1)Tarefa</h3>
<p>É possivel carregar um arquivo compactado usando o parametro compression no read do pandas e colocando o tipo do arquivo</p>

In [5]:
acidentes2018 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/MaC6cieXSFACNWT/download',
                            encoding='latin1',
                            sep=';',
                            low_memory=False, compression='zip')
acidentes2018.shape 

id data_inversa     dia_semana   horario  uf     br     km  \
0       99973.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
1       99976.0   2018-01-01  segunda-feira  00:40:00  SC  282.0    0,4   
2       99977.0   2018-01-01  segunda-feira  00:30:00  RJ  493.0      1   
3       99981.0   2018-01-01  segunda-feira  01:15:00  RS  386.0    134   
4       99982.0   2018-01-01  segunda-feira  00:20:00  RS  293.0  151,7   
...         ...          ...            ...       ...  ..    ...    ...   
69290  252256.0   2018-12-24  segunda-feira  09:15:00  PI  343.0  263,1   
69291  252652.0   2018-12-12   quarta-feira  07:00:00  BA  101.0    321   
69292  252801.0   2018-11-15   quinta-feira  10:48:00  SP  116.0    529   
69293  253040.0   2018-11-23    sexta-feira  19:30:00  RS  472.0  573,5   
69294  253078.0   2018-10-01  segunda-feira  04:31:00  GO   40.0      9   

                       municipio  \
0                        RESENDE   
1                  FLORIANOPOLIS   
2                       ITABORAI   
3                        SARANDI   
4                       CANDIOTA   
...                          ...   
69290                CAMPO MAIOR   
69291  PRESIDENTE TANCREDO NEVES   
69292             BARRA DO TURVO   
69293                 URUGUAIANA   
69294                   LUZIANIA   

                                          causa_acidente  \
0                                      Condutor Dormindo   
1                     Não guardar distância de segurança   
2                                 Ultrapassagem Indevida   
3                                     Ingestão de Álcool   
4                            Falta de Atenção à Condução   
...                                                  ...   
69290                                   Agressão Externa   
69291                        Falta de Atenção à Condução   
69292                        Falta de Atenção à Condução   
69293  Desobediência às normas de trânsito pelo condutor   
69294                 Não guardar distância de segurança   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0         Saída de leito carroçável  ...              4      1         0   
1                  Colisão traseira  ...              2      1         0   
2                   Colisão frontal  ...              0      3         1   
3               Colisão transversal  ...              0      2         0   
4         Saída de leito carroçável  ...              1      0         0   
...                             ...  ...            ...    ...       ...   
69290   Colisão com objeto estático  ...              0      1         0   
69291     Saída de leito carroçável  ...              0      0         0   
69292              Colisão traseira  ...              0      0         1   
69293  Queda de ocupante de veículo  ...              1      1         0   
69294     Saída de leito carroçável  ...              0      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1     -22,46937     -44,44705     SR-RJ     DEL5/7   
1           2        2  -27,59971687  -48,57565694     SR-SC     DEL8/1   
2           1        3    -22,763901    -42,927532     SR-RJ     DEL5/4   
3           0        2    -27,953636    -52,916374     SR-RS    DEL9/14   
4           1        1    -31,395214  -53,78391174     SR-RS    DEL9/11   
...       ...      ...           ...           ...       ...        ...   
69290       0        1     -4,819326    -42,158821     SR-PI    DEL17/1   
69291       1        1  -13,44768518  -39,42066193     SR-BA    DEL10/5   
69292       1        2    -24,939211    -48,301248     SR-SP     DEL6/5   
69293       1        2  -29,76355779  -57,05200195     SR-RS    DEL9/13   
69294       1        1    -16,131158    -47,957739     SR-DF   UOP01/DF   

            uop  
0      UOP03/RJ  
1      UOP01/SC  
2      UOP02/RJ  
3      UOP01/RS  
4      UOP03/RS  
...         ...  
69290  UOP03/PI  


In [4]:
acidentes2019 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/kRBUylqz6DyQznN/download',
                            encoding='latin1',
                            sep=';',
                            low_memory=False, compression='zip')
acidentes2019.shape

(67446, 30)

In [5]:
acidentes2020 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/jdDLrQIf33xXSCe/download',
                            encoding='latin1',
                            sep=';',
                            low_memory=False, compression='zip')
acidentes2020.shape

(63548, 30)

Concatenando os DataFrames

In [6]:
#visto que as colunas sao iguais em ambos dataframes vou usar o concat para colocar um em baixo do outro
todos_acidentes=pd.concat([acidentes2018,acidentes2019, acidentes2020], ignore_index=True)
todos_acidentes

id data_inversa     dia_semana   horario  uf     br     km  \
0        99973.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
1        99976.0   2018-01-01  segunda-feira  00:40:00  SC  282.0    0,4   
2        99977.0   2018-01-01  segunda-feira  00:30:00  RJ  493.0      1   
3        99981.0   2018-01-01  segunda-feira  01:15:00  RS  386.0    134   
4        99982.0   2018-01-01  segunda-feira  00:20:00  RS  293.0  151,7   
...          ...          ...            ...       ...  ..    ...    ...   
200284  364615.0   2020-03-20    sexta-feira  09:00:00  RN  304.0  171,8   
200285  364810.0   2020-09-12         sábado  23:50:00  BA  116.0   11,5   
200286  364817.0   2020-10-30    sexta-feira  10:30:00  BA  116.0   14,3   
200287  364835.0   2020-09-20        domingo  23:00:00  BA  116.0      8   
200288  364903.0   2020-10-31         sábado  02:10:00  BA  116.0      3   

                   municipio                      causa_acidente  \
0                    RESENDE                   Condutor Dormindo   
1              FLORIANOPOLIS  Não guardar distância de segurança   
2                   ITABORAI              Ultrapassagem Indevida   
3                    SARANDI                  Ingestão de Álcool   
4                   CANDIOTA         Falta de Atenção à Condução   
...                      ...                                 ...   
200284               ANGICOS         Falta de Atenção à Condução   
200285  VITORIA DA CONQUISTA                  Ingestão de Álcool   
200286  VITORIA DA CONQUISTA         Falta de Atenção à Condução   
200287  VITORIA DA CONQUISTA             Velocidade Incompatível   
200288  VITORIA DA CONQUISTA                    Animais na Pista   

                       tipo_acidente  ... feridos_graves ilesos ignorados  \
0          Saída de leito carroçável  ...              4      1         0   
1                   Colisão traseira  ...              2      1         0   
2                    Colisão frontal  ...              0      3         1   
3                Colisão transversal  ...              0      2         0   
4          Saída de leito carroçável  ...              1      0         0   
...                              ...  ...            ...    ...       ...   
200284              Colisão traseira  ...              1      2         0   
200285               Colisão lateral  ...              0      2         0   
200286               Colisão lateral  ...              0      2         0   
200287  Queda de ocupante de veículo  ...              0      0         0   
200288       Atropelamento de Animal  ...              1      0         0   

       feridos veiculos      latitude     longitude  regional  delegacia  \
0            4        1     -22,46937     -44,44705     SR-RJ     DEL5/7   
1            2        2  -27,59971687  -48,57565694     SR-SC     DEL8/1   
2            1        3    -22,763901    -42,927532     SR-RJ     DEL5/4   
3            0        2    -27,953636    -52,916374     SR-RS    DEL9/14   
4            1        1    -31,395214  -53,78391174     SR-RS    DEL9/11   
...        ...      ...           ...           ...       ...        ...   
200284       2        4   -5,68711053    -36,419458     SR-RN      UOP03   
200285       1        2  -14,87308214  -40,87141514     SR-BA      UOP01   
200286       2        2  -14,89597653  -40,86098671     SR-BA      UOP01   
200287       1        1  -14,84732305  -40,88634968     SR-BA      UOP01   
200288       1        1    -14,824848    -40,861311     SR-BA      UOP01   

             uop  
0       UOP03/RJ  
1       UOP01/SC  
2       UOP02/RJ  
3       UOP01/RS  
4       UOP03/RS  
...          ...  
200284     DEL01  
200285     DEL08  
200286     DEL08  
200287     DEL08  
200288     DEL08  

[200289 rows x 30 columns]

2ª Tarefa<br>
Explore os conjuntos de dados, respondendo as questões:<br>
(a) quantos acidentes ocorreram em cada ano?<br>
(b) quantas variáveis cada conjunto de dados registra?

<h3>2 Tarefa</h3>
<h4>a)</h4>Quantidades de acidentes em 2018, 2019 e 2020 respectivamente: 69295, 67446 e 63548.
<p><h4>b)</h4>Se cada coluna é uma variavel entao armazena 30 variaveis.</p>


In [7]:
acidentes2018=acidentes2018.assign(acidentes=1)
acidentes2019=acidentes2019.assign(acidentes=1)
acidentes2020=acidentes2020.assign(acidentes=1)

qtd_acid18=acidentes2018['acidentes'].sum()
qtd_acid19=acidentes2019['acidentes'].sum()
qtd_acid20=acidentes2020['acidentes'].sum()
print(qtd_acid18)
print(qtd_acid19)
print(qtd_acid20)

69295
67446
63548


In [8]:
#criando uma nova coluna com valor padrao 1 para somar
todos_acidentes=todos_acidentes.assign(acidentes=1)

(c) quais as cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais?

<h3>2)</h3>
<h4>c)</h4> Cidades que correram mais acidentes:

In [9]:

top5cidades=todos_acidentes[['municipio','acidentes']].groupby('municipio').sum()

top5cidades=top5cidades.sort_values(['acidentes'], ascending=False)

top5cidades.head(5)

acidentes
municipio           
CURITIBA        3063
BRASILIA        2997
SAO JOSE        2322
GUARULHOS       2111
PALHOCA         1887

(d) quantos acidentes com feridos graves aconteceram na Paraíba em 2019?

<h3>2)</h3><h4>d)</h4>Na paraiba no ano de 2019 tivemos 414 acidentes com feridos graves.

In [10]:
acidentesPB=acidentes2019.loc[acidentes2019['uf'] == 'PB']

In [11]:
acidentesPB.loc[acidentesPB['feridos_graves'] >= 1]

id data_inversa     dia_semana   horario  uf     br    km  \
47     182271.0   2019-01-01    terça-feira  04:15:00  PB  361.0  38,8   
82     182312.0   2019-01-01    terça-feira  13:45:00  PB  230.0  18,9   
357    182656.0   2019-01-03   quinta-feira  07:50:00  PB  101.0  81,3   
564    183385.0   2019-01-04    sexta-feira  13:15:00  PB  230.0  17,8   
950    183831.0   2019-01-06        domingo  21:25:00  PB  101.0    10   
...         ...          ...            ...       ...  ..    ...   ...   
66852  259567.0   2019-12-29        domingo  21:00:00  PB  230.0    27   
66868  259583.0   2019-12-29        domingo  23:20:00  PB  230.0     9   
66882  259597.0   2019-12-30  segunda-feira  03:50:00  PB  101.0  83,5   
66954  259678.0   2019-12-30  segunda-feira  15:35:00  PB  230.0   505   
67050  259780.0   2019-12-31    terça-feira  00:40:00  PB  101.0   122   

         municipio                                     causa_acidente  \
47     CATINGUEIRA                                 Ingestão de Álcool   
82     JOAO PESSOA                        Defeito Mecânico no Veículo   
357         BAYEUX  Desobediência às normas de trânsito pelo condutor   
564    JOAO PESSOA                 Não guardar distância de segurança   
950     MAMANGUAPE                        Falta de Atenção à Condução   
...            ...                                                ...   
66852  JOAO PESSOA                                   Animais na Pista   
66868     CABEDELO                        Falta de Atenção à Condução   
66882       BAYEUX                                 Ingestão de Álcool   
66954   CAJAZEIRAS                                         Mal Súbito   
67050      CAAPORA                            Velocidade Incompatível   

                      tipo_acidente  ... ilesos ignorados feridos veiculos  \
47        Saída de leito carroçável  ...      0         0       1        1   
82                       Tombamento  ...      1         0       1        2   
357                 Colisão lateral  ...      1         0       1        2   
564                Colisão traseira  ...      1         0       2        2   
950                Colisão traseira  ...      2         1       1        2   
...                             ...  ...    ...       ...     ...      ...   
66852       Atropelamento de Animal  ...      2         0       2        3   
66868     Atropelamento de Pedestre  ...      0         1       1        1   
66882              Colisão traseira  ...      0         1       2        2   
66954  Queda de ocupante de veículo  ...      0         0       1        1   
67050              Colisão traseira  ...      1         0       1        2   

          latitude     longitude regional  delegacia       uop  acidentes  
47     -7,10157405  -37,60036469    SR-PB    DEL14/3  UOP01/PB          1  
82       -7,129088    -34,850031    SR-PB    DEL14/1  UOP00/PB          1  
357      -7,126998    -34,941341    SR-PB    DEL14/1  UOP00/PB          1  
564    -7,12064741  -34,84509414    SR-PB    DEL14/1  UOP00/PB          1  
950        -6,5959      -35,1448    SR-PB    DEL14/1  UOP03/PB          1  
...            ...           ...      ...        ...       ...        ...  
66852      -7,1677      -34,8855    SR-PB    DEL14/1  UOP01/PB          1  
66868    -7,047271     -34,84844    SR-PB    DEL14/1  UOP01/PB          1  
66882     -7,13377     -34,92964    SR-PB    DEL14/1  UOP01/PB          1  
66954      -6,8995      -38,5494    SR-PB    DEL14/3  UOP03/PB          1  
67050    -7,450382     -34,97994    SR-PB    DEL14/1  UOP01/PB          1  

[414 rows x 31 columns]

3ª Tarefa<br>
Para cada indicação abaixo, construa um novo dataframe, salvando-o com a terminação .csv.<br>
(a) ranking de acidentes por estado para todos os anos (um só arquivo combinando a<br>
informação dos três anos);


<h3>3)Tarefa</h3>
<p><h4>a)</h4>criando e salvando dataframe com o ranking por Estado</p>

In [12]:
rankingUF=todos_acidentes[['uf','acidentes']].groupby('uf').sum()
rankingUF=rankingUF.sort_values(['acidentes'], ascending=False)
rankingUF

acidentes
uf           
MG      26160
SC      24145
PR      22850
RJ      13417
RS      13216
SP      12936
BA      10482
GO      10046
ES       8018
PE       8011
MT       6963
CE       4962
RO       4648
PB       4553
MS       4505
RN       4190
PI       3920
MA       3501
PA       3040
DF       2997
AL       1948
SE       1746
TO       1704
AC        789
RR        723
AP        456
AM        363

In [13]:
rankingUF.to_csv('rankingUF.csv',  header=True)

(b) acidentes por dia da semana para todos os anos (um só arquivo combinando a informação dos três anos)

<h3>3)</h3><h4>b)</h4>Criando o dataframe e salvando-o em .csv


In [14]:
rankingPorDia=todos_acidentes[['dia_semana','acidentes']].groupby('dia_semana').sum()
rankingPorDia=rankingPorDia.sort_values(['acidentes'], ascending=False)
rankingPorDia


acidentes
dia_semana              
sábado             33005
domingo            33001
sexta-feira        30777
segunda-feira      27075
quinta-feira       26229
quarta-feira       25353
terça-feira        24849

In [15]:
#salvando
rankingPorDia.to_csv('rankingUF.csv',  header=True)

4ª Tarefa<br>
Com relação a ocorrência dos acidentes, responda:<br>
(a) qual a causa mais frequente e a mais rara de acidentes registrados nos três anos?

<h3>4)Tarefa</h3><h4>a)</h4>Causa mais frequente: Falta de atencao a Conducao
<p>Causas mais raras: Acumulo de areia ou detritos sobre o pavimento,
Acostamento em desnível,
Acumulo de óleo sobre o pavimento,
Condutor usando celular,
Pista esburacada,
Ausência de sinalização,
Pedestre cruzava a pista fora da faixa,
Ingestão de álcool ou de substâncias psicoativas pelo pedestre,
Curva acentuada,
Fumaça,
Frear bruscamente,
Estacionar ou parar em local proibido</p><p>Obs: todas as causas anteriores tiveram apenas 1 ocorrencia dentro dos três anos</p>

In [27]:
causa_frequente=todos_acidentes[['causa_acidente','acidentes']].groupby('causa_acidente').sum()
causa_frequente.sort_values(['acidentes'], ascending=False)

acidentes
causa_acidente                                               
Falta de Atenção à Condução                             73231
Desobediência às normas de trânsito pelo condutor       24089
Velocidade Incompatível                                 18763
Ingestão de Álcool                                      15689
Não guardar distância de segurança                      12082
Defeito Mecânico no Veículo                             11190
Condutor Dormindo                                        7225
Pista Escorregadia                                       6752
Falta de Atenção do Pedestre                             5685
Animais na Pista                                         4551
Ultrapassagem Indevida                                   3590
Defeito na Via                                           3059
Avarias e/ou desgaste excessivo no pneu                  2712
Mal Súbito                                               1976
Restrição de Visibilidade                                1515
Objeto estático sobre o leito carroçável                 1422
Agressão Externa                                         1082
Desobediência às normas de trânsito pelo pedestre        1053
Carga excessiva e/ou mal acondicionada                   1049
Ingestão de álcool e/ou substâncias psicoativas...       1048
Sinalização da via insuficiente ou inadequada             780
Fenômenos da Natureza                                     761
Deficiência ou não Acionamento do Sistema de Il...        533
Ingestão de Substâncias Psicoativas                       253
Reação tardia ou ineficiente do condutor                   24
Condutor deixou de manter distância do veículo ...         21
Ausência de reação do condutor                             21
Manobra de mudança de faixa                                19
Transitar na contramão                                     14
Acessar a via sem observar a presença dos outro...         13
Demais falhas mecânicas ou elétricas                       13
Desrespeitar a preferência no cruzamento                    8
Ingestão de álcool pelo condutor                            8
Chuva                                                       6
Conversão proibida                                          6
Trafegar com motocicleta (ou similar) entre as ...          5
Pedestre andava na pista                                    4
Condutor desrespeitou a iluminação vermelha do ...          4
Acesso irregular                                            3
Transitar no acostamento                                    3
Falta de acostamento                                        3
Afundamento ou ondulação no pavimento                       2
Demais falhas na via                                        2
Iluminação deficiente                                       2
Mal súbito do condutor                                      2
Entrada inopinada do pedestre                               2
Acumulo de água sobre o pavimento                           2
Acumulo de areia ou detritos sobre o pavimento              1
Acostamento em desnível                                     1
Acumulo de óleo sobre o pavimento                           1
Condutor usando celular                                     1
Pista esburacada                                            1
Ausência de sinalização                                     1
Pedestre cruzava a pista fora da faixa                      1
Ingestão de álcool ou de substâncias psicoativa...          1
Curva acentuada                                             1
Fumaça                                                      1
Frear bruscamente                                           1
Estacionar ou parar em local proibido                       1

(b) qual a proporção de pessoas ilesas e de feridos graves por mês em cada um dos anos<br>
considerados? (proporção = numero de ilesos ou feridos graves/ número total de<br>
pessoas envolvidas no acidente);


<h3>4)</h3>
<h4>b)</h4>
<p>Proporção de pessoas ilesas em 2018 foi de 41,68%</p>
<p>Proporcao de pessoas feridas graves em 2018 foi de 9,99%</p>

In [53]:
total_envolvidos2018=acidentes2018[['ilesos','feridos','ignorados', 'feridos_graves']].sum() #agrupar total de ileses, feridos, ignorados e feridos graves
total_envolvidos2018=total_envolvidos2018.sum() #somar o total
total_envolvidos2018

177237

In [54]:
total_ilesos2018=acidentes2018['ilesos'].sum() #somar todos os ilesos
total_ilesos2018

73876

In [61]:
proporcao_ilesos2018=total_ilesos2018/total_envolvidos2018
proporcao_ilesos2018

0.41682041560170846

In [63]:
total_feridos2018=acidentes2018['feridos_graves'].sum() #somar todos os feridos graves
total_feridos2018

17706

In [65]:
proporcao_feridos2018=total_feridos2018/total_envolvidos2018
proporcao_feridos2018

0.09990013371925727

<h3>4)</h3>
<h4>b)</h4>
<p>Proporção de pessoas ilesas em 2019 foi de 39,1%</p>
<p>Proporcao de pessoas feridas graves em 2019 foi de 10,58%</p>

In [55]:
total_envolvidos2019=acidentes2019[['ilesos','feridos','ignorados', 'feridos_graves']].sum() 
total_envolvidos2019=total_envolvidos2019.sum() 
total_envolvidos2019

175513

In [56]:
total_ilesos2019=acidentes2019['ilesos'].sum()
total_ilesos2019

68634

In [58]:
proporcao_ilesos2019=total_ilesos2019/total_envolvidos2019
proporcao_ilesos2019

0.3910479565616222

In [59]:
total_feridos2019=acidentes2019['feridos_graves'].sum()
total_feridos2019

18573

In [66]:
proporcao_feridos2019=total_feridos2019/total_envolvidos2019
proporcao_feridos2019

0.10582122122008056

<h3>4)</h3>
<h4>b)</h4>
<p>Proporção de pessoas ilesas em 2020 foi de 38,15%</p>
<p>Proporcao de pessoas feridas graves em 2020 foi de 10,72%</p>

In [67]:
total_envolvidos2020=acidentes2020[['ilesos','feridos','ignorados', 'feridos_graves']].sum() 
total_envolvidos2020=total_envolvidos2020.sum() 
total_envolvidos2020

159491

In [68]:
total_ilesos2020=acidentes2020['ilesos'].sum()
total_ilesos2020

60858

In [69]:
proporcao_ilesos2020=total_ilesos2020/total_envolvidos2020
proporcao_ilesos2020

0.38157638989033865

In [71]:
total_feridos2020=acidentes2020['feridos_graves'].sum()
total_feridos2020

17104

In [72]:
proporcao_feridos2020=total_feridos2020/total_envolvidos2020
proporcao_feridos2020

0.10724116094325072

(c) mostre, reorganizando o conjunto de dados, se a pandemia conseguiu diminuir ou não
a incidência de acidentes nas rodovias federais.

<h3>4)</h3>
<h4>c)</h4>
<p>No ano de 2020 em comparação a quantidade de acidentes em 2019 a diferença é de 3898 acidentes a mais em 2020. Já em relação à 2018 a diferença é de 5747.</p>
<p>Os acidentes diminuiram aproximadamente 7% em relação a média de acidentes dos anos anteriores.</p>

In [74]:
view18=acidentes2018['acidentes'].sum()
view18

69295

In [75]:
view19=acidentes2019['acidentes'].sum()
view19

67446

In [76]:
view20=acidentes2020['acidentes'].sum()
view20

63548

In [77]:
media=(view18+view19)/2
media

68370.5

In [78]:
razao=view20/media
razao #92,94%

0.9294651933216811